pkg/background/common/context.go

```diff
@@ -24,57 +25,57 @@ func NewBackgroundContext(dclient dclient.Interface, ur *urkyverno.UpdateRequest
 	ctx := context.NewContext()
 	requestString := ur.Spec.Context.AdmissionRequestInfo.AdmissionRequest
 	var request admissionv1.AdmissionRequest
+	var new, old unstructured.Unstructured
 
-	err := json.Unmarshal([]byte(requestString), &request)
-	if err != nil {
-		logger.Error(err, "error parsing the request string")
-	}
+	if requestString != "" {
+		err := json.Unmarshal([]byte(requestString), &request)
+		if err != nil {
+			return nil, false, errors.Wrap(err, "error parsing the request string")
+		}
 
-	if err := ctx.AddRequest(&request); err != nil {
-		logger.Error(err, "failed to load request in context")
-		return nil, false, err
-	}
+		if err := ctx.AddRequest(&request); err != nil {
+			return nil, false, errors.Wrap(err, "failed to load request in context")
+		}
 
-	new, old, err := utils.ExtractResources(nil, &request)
-	if err != nil {
-		logger.Error(err, "failed to load request in context")
-		return nil, false, err
-	}
+		new, old, err = utils.ExtractResources(nil, &request)
+		if err != nil {
+			return nil, false, errors.Wrap(err, "failed to load request in context")
+		}
 
-	if !reflect.DeepEqual(new, unstructured.Unstructured{}) {
-		if !check(&new, trigger) {
-			err := fmt.Errorf("resources don't match")
-			logger.Error(err, "", "resource", ur.Spec.Resource)
-			return nil, false, err
+		if !reflect.DeepEqual(new, unstructured.Unstructured{}) {
+			if !check(&new, trigger) {
+				err := fmt.Errorf("resources don't match")
+				return nil, false, errors.Wrapf(err, "resource %v", ur.Spec.Resource)
+			}
 		}
 	}
 
 	if trigger == nil {
 		trigger = &old
 	}
 
-	err = ctx.AddResource(trigger.Object)
+	if trigger == nil {
+		return nil, false, errors.New("trigger resource does not exist")
+	}
+
```

Comment:
return statement can be move to above `trigger==nil{}` check ?

Reply:
This scenario happens when the policy is created but the trigger does not exist.

The first check tries to assign trigger by `old` which is parsed from the admission request (DELETE request), and this check returns if the admission request is empty (which means this is on policy creation without the trigger exists in the cluster).

Reply:
got it .